source-kinesis/connection.go

```diff
@@ -12,12 +12,16 @@ import (
 )
 
 // Config represents the fully merged endpoint configuration for Kinesis.
-// It matches the `KinesisConfig` struct in `crates/sources/src/specs.rs`
 type Config struct {
-	Endpoint           string `json:"endpoint,omitempty" jsonschema:"title=AWS Endpoint,description=The AWS endpoint URI to connect to (useful if you're capturing from a kinesis-compatible API that isn't provided by AWS)"`
 	Region             string `json:"region" jsonschema:"title=AWS Region,description=The name of the AWS region where the Kinesis stream is located"`
 	AWSAccessKeyID     string `json:"awsAccessKeyId" jsonschema:"title=AWS Access Key ID,description=Part of the AWS credentials that will be used to connect to Kinesis"`
 	AWSSecretAccessKey string `json:"awsSecretAccessKey" jsonschema:"title=AWS Secret Access Key,description=Part of the AWS credentials that will be used to connect to Kinesis"`
+
+	Advanced advancedConfig `json:"advanced,omitempty"`
+}
+
+type advancedConfig struct {
+	Endpoint string `json:"endpoint,omitempty" jsonschema:"title=AWS Endpoint,description=The AWS endpoint URI to connect to (useful if you're capturing from a kinesis-compatible API that isn't provided by AWS)"`
```

Comment:
Moving this property into the advanced config will break any existing configs. Are we prepared for whatever breakage that might cause?

Reply:
Yep there aren't any real usages of this connector yet, any the only two tasks that exist (one is our own testing task, and another is a disabled & probably abandoned one) don't have this field set so it shouldn't actually break them.